AndroidNativeLoginTemplate/app/src/main/java/com/salesforce/androidnativelogintemplate/NativeLogin.kt

```diff
@@ -0,0 +1,327 @@
+                        nativeLoginManager.getFallbackWebAuthenticationIntent(),
+                        nativeLoginManager.shouldShowBackButton,
+                        backAction = { finish() },
+                    )
+                }
+            }
+        )
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            onBackInvokedDispatcher.registerOnBackInvokedCallback(
+                OnBackInvokedDispatcher.PRIORITY_DEFAULT
+            ) {
+                if (nativeLoginManager.shouldShowBackButton) {
+                    when (SalesforceSDKManager.getInstance().userAccountManager.authenticatedUsers) {
+                        null -> moveTaskToBack(false)
+                        else -> finish()
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun LoginView(
+    login: suspend (String, String) -> Boolean,
+    handleWebviewFallbackResult: ActivityResultLauncher<Intent>? = null,
+    webviewLoginIntent: Intent? = null,
+    shouldShowBack: Boolean = false,
+    backAction: () -> Unit,
+) {
+    LoginTheme {
+        Scaffold(
+            topBar = {
+                Row(modifier = Modifier.statusBarsPadding()) {
+
+                    // Back button should only be shown if there is a user already logged in.  But not
+                    // in the case of Biometric Authentication.
+                    if (shouldShowBack) {
+                        TextButton(onClick = { backAction() }) {
+                            Image(
+                                painter = painterResource(id = com.salesforce.androidsdk.R.drawable.sf__action_back),
+                                colorFilter = ColorFilter.tint(MaterialTheme.colorScheme.primary),
+                                contentDescription = "Back",
+                                modifier = Modifier.padding(start = 16.dp)
+                            )
+                        }
+                    }
+                }
+            },
+            bottomBar = {
+                Column(
+                    horizontalAlignment = Alignment.CenterHorizontally,
+                    modifier = Modifier
+                        .navigationBarsPadding()
+                        .fillMaxWidth(),
+                ) {
+                    // Fallback to web based authentication.
+                    TextButton(onClick = { handleWebviewFallbackResult?.launch(webviewLoginIntent) }) {
+                        Text(text = "Looking for Salesforce Log In?")
+                    }
+                }
+            },
+        ) { innerPadding ->
+            Column(
+                horizontalAlignment = Alignment.CenterHorizontally,
+                modifier = Modifier
+                    .verticalScroll(rememberScrollState(), reverseScrolling = true)
+                    .navigationBarsPadding()
+                    .padding(top = innerPadding.calculateTopPadding())
+                    .imePadding()
```

Comment:
`imePadding()` here (vs on the text views) with `verticalScroll` makes it such that if you tap the username field the entire card shits up enough that the password field and login button are not hidden by the keyboard 😎